### PR TITLE
Remove the rule of force adding spaces

### PR DESCRIPTION
**Pull request by zpengyu**
Rules in this section applied to all different languages we are currently using. I don't think this rule should apply to all situations.

For example,
* We don't add space around curly brackets in coffeescript sometimes.
* Sometimes we add multiple spaces around `=` to align vertically.
* You can find some lines of code in this file: https://github.com/studentcompetitions/sqore/blob/master/app/assets/javascripts/sqore/services/notifications.js.coffee.erb

Another example,
`x*y + x*y` is easier to read than `x * y + x * y`
`(x+y) * (x+y)` is easier to read than `( x + y ) * ( x + y )`

My point is not about removing all spaces, but add them properly as long as `y o u r  c o d e` is easy to understand.

### CODE REVIEW

**Comment by chav02** [comment]
I happen to like that rule except perhaps for around certain operators.
`{ |s| s.method({ a:b }) { |n| t = n }` does look better than `{|s|s.method({a:b}){|n|t=n}`

also this rule does not ask for spaces inside (), in fact current rubocop settings forbids it.

perhaps add wording to exclude `*` and any other specific operators?

**Comment by SteveLTN** [comment]
Normally I would add spaces around `{` and `}` , etc.

But in certain cases, e.g. `{[`, `]}` I see the point.

**Comment by Frost** [comment]
I agree with @zpengyu on the vertical align thing, but not the rest of it.

I like this rule, except the vertical align bit, so I don't think we should implement these changes.

**Comment by Koronen** [comment]
I agree with @chav02 @SteveLTN and @Frost.

**Comment by Koronen** [comment]
Christian:
> perhaps add wording to exclude * and any other specific operators?

Martin:
> I agree with @zpengyu on the vertical align thing, but not the rest of it.

---

@zpengyu update the wording a bit, maybe?

**Comment by zpengyu** [comment]
Ping @studentcompetitions/developers

File: style/README.md [hunk 1]
```diff
@@ -29,8 +29,8 @@ Formatting
 * Use 2 space indentation (no tabs).
 * Use an empty line between methods.
 * Use newlines around multi-line blocks.
-* Use spaces around operators, after commas, after colons and semicolons, around
-  `{` and before `}`.
+* Prefer to use spaces around operators, after commas, after colons and semicolons, around
+  `{` and before `}`. None or extra spaces are acceptable when improving readability.
```
Comment (Koronen) [inline style/README.md]: From README.md:

> High level guidelines:
> * Don't violate a guideline without a good reason.
> * A reason is good when you can convince a teammate.
>
> A note on the language:
> * "Prefer" indicates a better option and its alternative to watch out for.

---

I think we can remove this last sentence, as the main README file already covers this.

**Comment by Koronen** [comment]
:+1: from me on this new wording.

**Comment by maxnordlund** [comment]
:+:
Den 3 mars 2016 14:29 skrev "Victor Koronen" <notifications@github.com>:

File: style/README.md (hunk 1 above)
Comment (Koronen) [inline style/README.md]: Thoughts on this?

Comment (Frost) [inline style/README.md]: Agreed.

**Comment by Koronen** [comment]
Close this due to age and lack of activity?

**Pull request closed without merging**